OpenSim/Common/Object.h

```diff
@@ -518,9 +518,13 @@ class OSIMCOMMON_API Object
         no parent node is supplied and this object doesn't already have an XML 
         node, this object will become the root node for a new XML document. If 
         this object already has an XML node associated with it, no new nodes 
-        are ever generated and the parent node is not used. 
-    **/
-    virtual void updateXMLNode(SimTK::Xml::Element& parent) const;
+        are ever generated and the parent node is not used.
+    @param      prop (optional)
+        The pointer to the property that contains this object. If it is
+        present, check if the property is unnamed and if NOT, use the property
+        name as its name when updating the XML node. **/
+    void updateXMLNode(SimTK::Xml::Element& parent,
+                       const AbstractProperty* prop=nullptr) const;
```

Comment:
I forget if I already asked this...is `prop` always passed? If so, I think we should make it a required argument. Otherwise, we could get different deserializations based on whether or not this parameter is passed.

Reply:
No it is not always passed- it is optional.  You previously asked to make it clear that it was optional. It is used exclusively to replace the KLUDGE in `AbstractProperty` that was renaming a `const Object`.

Reply:
Thanks for clarifying. It's been difficult for me to keep track.